File: api/Models/Product.cs
using System.ComponentModel.DataAnnotations;

namespace api.Models
{
    public class Product
    {
        public int ProductId { get; set; }
        [Required]
        public string Name { get; set; }
        [Required]
        public string Image { get; set; }
        //Relation
        public int CategoryId { get; set; }
        public Category Category { get; set; }
        public List<Ingredient> Ingredients { get; set; } = new List<Ingredient>();
        public List<ProductItem> ProductItems { get; set; } = new List<ProductItem>();

    }
}

File: frontend/src/Shared/Models/Product.ts
import { IngredientForProducts } from "./Ingredient"
import { ProductItemGet } from "./ProductItem"

export type ProductGetSearch = {
    productId: number,
    name:string,
    image:string,
}

export type ProductShortGet = {
    productId: number,
    name:string,
    image:string,
}

export type ProductGetWithIngredientsWithItems={
    productId: number,
    name:string,
    image:string,
    ingredients:IngredientForProducts[]
    productItems:ProductItemGet[],
}


